src/bundle/Resources/public/js/alloyeditor/src/toolbars/config/base-fixed.js

```diff
@@ -0,0 +1,56 @@
+import EzConfigBase from './base';
+
+let editorialInteractionListener = null;
+let blurListener = null;
+
+export default class EzConfgiFixedBase extends EzConfigBase {
+    static getTopPosition(block, editor) {
+        const toolbar = document.querySelector('.ae-toolbar-floating');
+        const editorRect = editor.element.getClientRect();
+        const toolbarHeight = toolbar.getBoundingClientRect().height;
+        const offset = 10;
+        const shouldBeFixed = editorRect.top - toolbarHeight - 2 * offset < 0;
+        const top = shouldBeFixed ? offset : editorRect.top + editor.element.getWindow().getScrollPosition().y - toolbarHeight - offset;
+
+        toolbar.classList.toggle('ae-toolbar-floating--fixed', shouldBeFixed);
```

Comment:
I wouldn't expect that function named `getTopPosition` would toggle CSS class on the element. In my opinion it should be moved somewhere else.

Reply:
Discussed is private.